learn/advanced/faceted_search.md

```diff
@@ -0,0 +1,227 @@
+
+## Facet types
+
+### Conjunctive facets
+
+Conjunctive facets use the `AND` logical operator. When users select one or more values for a facet, all returned results must contain the selected facet value(s).
+
+With conjunctive facets, when a user selects `English` from the `language` facet, all returned books must be in English. If the user further narrows down the search by selecting `Fiction` and `Literature` as `genres`, all returned books must be in English and contain both `genres`.
+
+```
+# Filter expression as an array
+["language = English", "genres = Fiction", "genres = Literature"]
+
+# Filter expression as a string
+"language = English AND genres = Fiction AND genres = Literature"
+```
+
+The GIF below shows how the facet count for `genres` updates to only include books that meet **all three conditions**.
+
+![Selecting English books with 'Fiction' and 'Literature' as 'genres' for the books dataset](/faceted-search/conjunctive-factes.gif)
+
+### Disjunctive facets
+
+Disjunctive facets use the `OR` logical operator. They allow users to choose multiple options within a facet, so they don’t have to perform more than one search to find the products they’re looking for.
+
+Let's look at the `books` index from before with disjunctive facets. When the user selects `Fiction` and `Literature` as `genres`, Meilisearch returns all books that are either `Fiction`, `Literature`, or both:
+
+```
+# Filter expression as an array
+[["genres = Fiction", "genres = Literature"]]
+
+# Filter expression as a string
+"genres = Fiction OR genres = Literature"
+```
+
+The GIF below shows the `books` dataset with disjunctive facets. Notice how the facet count for `genres` updates based on the selection.
+
+![Selecting 'Fiction' and 'Literature' as 'genres' for the books dataset](/faceted-search/disjunctive_facets.gif)
+
+### Using conjunctive and disjunctive facets
+
+Let's look at the `books` index with both conjunctive and disjunctive facets. The user selects `English` and `French` from the `language` facet, and then narrows down the search to include `Fiction` and `Literature` as `genres`:
+
+```
+# Filter expression as an array
+[["language = English", "language = French"], "genres = Fiction", "genres = Literature"]
+
+# Filter expression as a string
+"(language = English OR language = French) AND (genres = Fiction AND genres = Literature)"
+```
+
+![Selecting 'Fiction' and 'Literature' as 'genres' for English books](/faceted-search/conjunctive-and-disjunctive-facets.gif)
+
+::: tip Improving user experience
+Most ecommerce websites use different ways for handling facet counts depending on their use case. By keeping the facet count visible, you allow the user to make more informed decisions. They can select or deselect different facets based on how many results they get for each facet and how those results overlap with other selected facets. This can help the user quickly refine their search results to find the most relevant items.
```

Comment:
> Most ecommerce websites use different ways for handling facet counts depending on their use case. By keeping the facet count visible, you allow the user to make more informed decisions.

Bit of a strange context switch between the two sentences. It should introduce `one of the best ways to handle it is by keeping the facet count visible [...]`

Reply:
This part of the chatGPT explaination is maybe a better introduction 

> Multi-select faceting can be designed in various ways, depending on the specific use case and the preferences of the users. One possible approach is to keep the facet count visible even after the user selects one or more facets. This means that the user can see how many results are associated with each facet, and how many of those results match the other facets that have been selected.